### PR TITLE
TMA-483 && TMA-963 Paralel ufb bug fix

### PR DESCRIPTION
**Pull request by Hahihula**

### CODE REVIEW

File: lib/gooddata/lcm/actions/synchronize_users.rb [hunk 1]
```diff
@@ -198,10 +198,10 @@ def call(params)
                                            do_not_touch_users_that_are_not_mentioned: do_not_touch_users_that_are_not_mentioned,
                                            create_non_existing_user_groups: create_non_existing_user_groups)
                     when 'sync_multiple_projects_based_on_pid'
-                      new_users.group_by { |u| u[:pid] }.flat_map do |project_id, users|
+                      new_users.group_by { |u| u[:pid] }.flat_map.pmap do |project_id, users|
                         begin
-                          project = client.projects(project_id)
-                          project.import_users(users,
+                          current_project = client.projects(project_id)
+                          current_project.import_users(users,
```
Comment (yenkins) [inline lib/gooddata/lcm/actions/synchronize_users.rb]: warning: Similar code found in synchronize_users.rb:193, synchronize_users.rb:204, synchronize_users.rb:221, synchronize_users.rb:247, synchronize_users.rb:265, synchronize_users.rb:292, synchronize_users.rb:334 (mass = 112) (flay)

Comment (yenkins) [inline lib/gooddata/lcm/actions/synchronize_users.rb]: info: Calls 'current_project.import_users(users, ... create_non_existing_user_groups: create_non_existing_user_groups)' 3 times (DuplicateMethodCall) (reek)

File: lib/gooddata/lcm/actions/synchronize_users.rb [hunk 2]
```diff
@@ -252,17 +252,17 @@ def call(params)
...
                         rescue RestClient::BadRequest => e
                           raise e unless /does not exist in data product/ =~ e.response
                           fail "The client \"#{client_id}\" does not exist in data product \"#{data_product.data_product_id}\""
                         end
-                        fail "Client #{client_id} does not have project." unless project
-                        puts "Project #{project.pid} of client #{client_id} will receive #{users.count} users"
-                        project.import_users(users,
+                        fail "Client #{client_id} does not have project." unless current_project
+                        puts "Project #{current_project.pid} of client #{client_id} will receive #{users.count} users"
+                        current_project.import_users(users,
```
Comment (yenkins) [inline lib/gooddata/lcm/actions/synchronize_users.rb]: warning: Similar code found in synchronize_users.rb:193, synchronize_users.rb:204, synchronize_users.rb:221, synchronize_users.rb:247, synchronize_users.rb:265, synchronize_users.rb:292, synchronize_users.rb:334 (mass = 112) (flay)

File: lib/gooddata/lcm/actions/synchronize_users.rb [hunk 3]
```diff
@@ -285,11 +285,11 @@ def call(params)
...
                         end
-                        project = c.project
-                        fail "Client #{client_id} does not have project." unless project
+                        current_project = c.project
+                        fail "Client #{client_id} does not have project." unless current_project
                         working_client_ids << client_id.to_s
-                        puts "Project #{project.pid} of client #{client_id} will receive #{users.count} users"
-                        project.import_users(users,
+                        puts "Project #{current_project.pid} of client #{client_id} will receive #{users.count} users"
+                        current_project.import_users(users,
```
Comment (yenkins) [inline lib/gooddata/lcm/actions/synchronize_users.rb]: warning: Similar code found in synchronize_users.rb:193, synchronize_users.rb:204, synchronize_users.rb:221, synchronize_users.rb:247, synchronize_users.rb:265, synchronize_users.rb:292, synchronize_users.rb:334 (mass = 112) (flay)

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/526/ : SUCCESS in 6s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1361/ : FAILURE in 1m 18s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1631/ : SUCCESS in 1m 10s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/528/ : SUCCESS in 6s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1363/ : FAILURE in 2m 45s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1633/ : SUCCESS in 1m 04s

**Pull request closed without merging**